libraries/botbuilder-dialogs-declarative/src/defaultLoader.ts

```diff
@@ -39,6 +39,9 @@ export class DefaultLoader implements CustomDeserializer<Configurable, Record<st
                 }
                 instance[`${key}`] = converter.convert(value);
             } else {
+                if (key === 'schema' && typeof value === 'string') {
+                    value = JSON.parse(this._resourceExplorer.getResource(value).readText());
+                }
                 instance[`${key}`] = value;
             }
 
```

Comment:
Note: Some of the `.dialog` test files were written like:

```json
{
   <normal .dialog JSON>,
   "schema": "someSchemaFile.json"
}
```

In .NET, it's smart enough to see that `someSchemaFile.json` and appropriately reads the file and deserializes it into an object:

```json
{
   <normal .dialog JSON>,
   "schema": {
		<someSchemaFiles's JSON>
	}
}
```

However, in JS this stays as the string, `someSchemaFile.json` and the file would never get read, so you couldn't have this nested `"schema"` property.

**TL;DR** I don't know if this is the best place to do this, but it made sense to me. In .NET, [`JObjectConverter`](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Converters/JObjectConverter.cs) eventually calls [`GetRefTarget()`](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs#L576) and knows to read the file and convert it.